### PR TITLE
Adopt Starter Pack config

### LINKED ISSUE #130: Use Canonical sphinx documentation pack – 2.0 Release (meta-issue)
This issue serves as a tracker for all issues and pull requests related to setting up and maintaining the use of the [Canonical Sphinx documentation pack](https://github.com/canonical/sphinx-docs-starter-pack) for the Ubuntu Packaging Guide as part of the 2.0 release (see #19).

### Motivation

- Ensure consistency with other Ubuntu/Canonical documentation.
- Modernize and improve the existing documentation.

### Scope

This issue covers tasks such as:

- Updating dependencies and configurations.
- Refining styles and templates for consistency.
- Addressing structural improvements and best practices.
- Resolving any related bugs or enhancement requests.

Feel free to link relevant issues and PRs here for better tracking.

### PR DESCRIPTION
**Pull request by rkratky**
- Adopts Starter Pack config wherever possible (unifies `conf.py`, moves custom config to `custom_conf.py`, uses the standard `Makefile`, `requirements.txt`, and `.readthedocs.yaml`, etc. etc.)
- Introduces the standard Canonical RTD style, incl. the header.

### CODE REVIEW

**Comment by dviererbe** [comment]
question: Do the docs/.sphinx/.doctree/** files need to be commited?

They look like caching files to me.

**Comment by dviererbe** [comment]
The changes to `docs/.sphinx/requirements.txt` is problematic for deb packaging as mentioned in the discussion in Riga, but I am fine to have them for now as long as we can get rid of them by the end of the cycle.

File: docs/reuse/links.txt
Comment (dviererbe) [inline docs/reuse/links.txt:1]: question: Can we please keep the `.rst` ending? Otherwise I have to manually set the parser in my editor to get syntax highlighting.

File: docs/.sphinx/requirements.txt
Comment (dviererbe) [inline docs/.sphinx/requirements.txt:1]: note: The added dependencies and changed sphinx version `sphinx<6.0dev` -> `sphinx` is problematic for deb packaging as mentioned in the discussion in Riga, but I am fine to have them for now as long as we can solve this problem by the end of the cycle.

File: docs/.sphinx/_static/furo_colors.css
Comment (dviererbe) [inline docs/.sphinx/_static/furo_colors.css:1]: note: This is a bit weird to me why they moved this into a css file, as the documentation for the furo theme recommends setting these variables in the `conf.py` and the `conf.py` should not be modified by the downstreams (like us) anyway.

File: docs/.sphinx/warnings.txt
Comment (dviererbe) [inline docs/.sphinx/warnings.txt:1]: question: Why do we need this file in version control?

As far as I can tell it is only used [here](https://github.com/rkratky/ubuntu-packaging-guide/blob/e0b646c39670d058204133b56b7048e284cb58de/docs/Makefile#L59-L63):
```make
html: install
	. $(VENV); $(SPHINXBUILD) -b dirhtml "$(SOURCEDIR)" "$(BUILDDIR)" -w .sphinx/warnings.txt $(SPHINXOPTS)

epub: install
	. $(VENV); $(SPHINXBUILD) -b epub "$(SOURCEDIR)" "$(BUILDDIR)" -w .sphinx/warnings.txt $(SPHINXOPTS)
```

and if I understands the [documentation](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-w) correctly, it should create the file by itself.

File: docs/Makefile
Comment (dviererbe) [inline docs/Makefile:46]: question: Why is this here?

I am aware that `$(VENVDIR)` is called by `help`, but when the folder already exists (aka after the first help call), this will not be displayed again. This looks like bad UX to me.

File: docs/conf.py
Comment (dviererbe) [inline docs/conf.py:30]: note: As already mentioned [here](https://github.com/canonical/ubuntu-packaging-guide/pull/43#discussion_r1403640690). This may be problematic for deb packaging.

File: docs/custom_conf.py
Comment (dviererbe) [inline docs/custom_conf.py]: issue: This should not be be linking to foundations. The product is Ubuntu Packaging/Development this is nothing foundations specific.

suggestion: Linking just to https://discourse.ubuntu.com/ should be enough:
```suggestion
    'discourse': 'https://discourse.ubuntu.com/',
```

File: docs/custom_conf.py
Comment (dviererbe) [inline docs/custom_conf.py]: nitpick: I am not sure if packages.ubuntu.com/ a good landing page for the "product". Launchpad may be better but also not ideal.

**Review by dviererbe — CHANGES_REQUESTED** [review]
praise: Thanks @rkratky for syncing with the starter pack :)

todo: I have some minor questions regarding the changes. I will hold back for approval until they are resolved.

**Comment by rkratky** [comment]
> question: Do the docs/.sphinx/.doctree/** files need to be commited?
> 
> They look like caching files to me.

Of course. `.gitignore` oversight.

File: docs/.sphinx/requirements.txt
Comment (rkratky) [reply docs/.sphinx/requirements.txt:1]: I'm sorry, could you please remind me what the problem is? I'm drawing a blank on this :(

File: docs/.sphinx/_static/furo_colors.css
Comment (rkratky) [reply docs/.sphinx/_static/furo_colors.css:1]: I don't think the Furo docs prescribe having it in `conf.py` as the only way -- they just mention that you can customize the theme using the file. But in our case (when it's a sizeable chunk of CSS code), I think it makes sense to have the CSS stuff in a separate CSS file (to keep `conf.py` nice and small and also because it just seems to make sense to keep CSS definitions in CSS files).

File: docs/.sphinx/warnings.txt
Comment (rkratky) [reply docs/.sphinx/warnings.txt:1]: Ack, it's just me overlooking it.

File: docs/custom_conf.py
Comment (rkratky) [reply docs/custom_conf.py]: Yeah, I wasn't sure either. Where in LP would you link? My reasoning was that packages are the desired outcome :)

File: docs/custom_conf.py
Comment (rkratky) [reply docs/custom_conf.py]: Ack.

File: docs/Makefile
Comment (rkratky) [reply docs/Makefile:46]: And badly formatted, too ;)

Anyway, I'm aware that the Starter Pack isn't perfect yet, but the idea is to unify on its use, so that future updates and docs maintenance is simple across the board. As we update the SP, changes will be propagated to projects that have adopted it.

IOW, would you be OK with raising this as an issue for the SP?

Comment (dviererbe) [reply docs/Makefile:46]: That's fine for me. :+1:

File: docs/custom_conf.py
Comment (dviererbe) [reply docs/custom_conf.py]: What do you think about the following?
1.) https://launchpad.net/ubuntu (Ubuntu project page on Launchpad)
2.a) https://ubuntu.com/community/contribute
2.b) https://ubuntu.com/community/contribute/ubuntu-development

File: docs/.sphinx/requirements.txt
Comment (dviererbe) [reply docs/.sphinx/requirements.txt:1]: We want to package the guide as a deb package. Therefore every dependency needs to be packaged. Some of the dependencies do not have a package yet. Therefore we would need to package and maintain the dependencies by our-self. This is a lot of work for many dependencies that we do not even use.

Also I think @waveform80 set `sphinx<6.0dev`, because it matches the packaged version in the archive (see https://launchpad.net/ubuntu/+source/sphinx) and the one in Debian `stable` (see https://tracker.debian.org/pkg/sphinx).

This does not need to be resolved with this PR, but it should be resolved by the end of the cycle. There already exists an issue for that upstream: https://github.com/canonical/sphinx-docs-starter-pack/issues/140

File: docs/custom_conf.py
Comment (rkratky) [reply docs/custom_conf.py]: But those are community and/or dev resources. This link, in my understanding, should point to the 'product' the docs talk about. I'd say if not packages directly, then the product is Ubuntu itself.

File: docs/.sphinx/requirements.txt
Comment (rkratky) [reply docs/.sphinx/requirements.txt:1]: Thanks for the reminder.

File: docs/custom_conf.py
Comment (dviererbe) [reply docs/custom_conf.py]: I think the "product" here is not packages of Ubuntu, but rather contributing (especially deb packaging related) to Ubuntu. Therefore I would argue that 2.b) makes the most sense. It also does not drop people into a random place with no further context. Any other Canonical product page also tells people what the product is and how you can use it. packages.ubuntu.com/ does neither.

But if you think that the "product" here is packages of Ubuntu then I think using 1) instead still makes sense. Launchpad is the platform where you interact with Ubuntu, its packages and its community.

Comment (s-makin) [reply docs/custom_conf.py]: Could be worth seeing if the Community team has any suggestions there. If we want to point people to general contribution resources, I think they have a bunch of Discourse pages that might be suitable.

File: docs/custom_conf.py
Comment (s-makin) [reply docs/custom_conf.py]: I'd probably flag this as another place where we might be able to link back to the Community team, who has their own Discourse. We should check with them what Discourse they think would be the best to drive traffic to before this goes live. Doesn't have to be resolved in this PR, but pointing just to the generic discourse.ubuntu might leave people feeling a bit confused about where they should be posting (the landing page for that discourse can be a bit overwhelming).

File: docs/reuse/links.txt
Comment (s-makin) [reply docs/reuse/links.txt:1]: In this case I think it's better not to. Anything with a .rst ending will be flagged up by Sphinx who will complain that it's not in any toc/navigation. We could add the `:orphan:` tag to the top of the file, but then it will often be ignored when we're trying to include it. Makes a bit of a mess, in other words :upside_down_face:.

File: docs/custom_conf.py
Comment (s-makin) [reply docs/custom_conf.py]: In the meantime, I think "ubuntu.com" could be the best option?

**Comment by s-makin** [comment]
> praise: Thanks @rkratky for syncing with the starter pack :)
> 
> todo: I have some minor questions regarding the changes. I will hold back for approval until they are resolved.

I can't see anything additional to what Dominik has already picked up on. Left a few comments of my own, but once all Dominik's points have been addressed I'd be happy to add my approval as well.

File: docs/custom_conf.py
Comment (rkratky) [reply docs/custom_conf.py]: > Launchpad is the platform where you interact with Ubuntu, its packages and its community.

I think the difference is who we expect to be the main consumer of the docs. You see a community developer who would like to become an Ubuntu packager and contribute to the project. I see a customer who wants to package their own software to facilitate deployment on their installations. I suppose both are valid, but ultimately, yours makes more sense because it is a superset of mine.

File: docs/reuse/links.txt
Comment (dviererbe) [reply docs/reuse/links.txt:1]: It didn't complain so far. This PR just renames the file.

This is just a personal preference, so I will not block on it.

Comment (rkratky) [reply docs/reuse/links.txt:1]: It's just a warning:
```
checking consistency... /docs/reuse/links.rst: WARNING: document isn't included in any toctree
```
Leaving with `.txt` for now to keep it consistent with the Starter Pack. Let's do the chanfge in the Starter Pack (if needed).

**Comment by rkratky** [comment]
Thank you both. @s-makin, @dviererbe, please, have one final look. I updated what was discussed above.

**Review by s-makin — APPROVED** [review]
LGTM, although I'll hold off merging in case Dominik has any further points he wants to add. 

One minor note - the favicon update you made to the starter pack is now live, so if you wanted to include it here as well it might make sense to do so (rather than adding it as a separate PR) but I'll leave that up to you to decide :)

**Comment by rkratky** [comment]
> One minor note - the favicon update you made to the starter pack is now live, so if you wanted to include it here as well it might make sense to do so (rather than adding it as a separate PR) but I'll leave that up to you to decide :)

Good point. I added it.

@dviererbe Could you please have one last look?

File: docs/reuse/links.txt
Comment (dviererbe) [reply docs/reuse/links.txt:1]: Okay, may bad. I did not remember to see a warning, but in that case the rename is a a good idea 👍🏻

File: docs/custom_conf.py
Comment (dviererbe) [inline docs/custom_conf.py:100]: Note: We should setup redirects for the 1.0 version (In a separate PR).

**Review by dviererbe — APPROVED** [review]
LGTM

**Pull request merged**
